### PR TITLE
executor: support batch commit the DMLs

### PR DESCRIPTION
**Pull request by jackysp**
### What problem does this PR solve? 
Make all the DMLs could be batch committed. 
The old version for release-2.0 will be closed. https://github.com/pingcap/tidb/pull/9228

### What is changed and how it works?
Add a config to control to split DMLs.

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change

Side effects

- Increased code complexity

### CODE REVIEW

**Comment by jackysp** [comment]
/run-all-tests tidb-test=release-2.1 tidb-private-test=release-2.1 tikv=release-2.1

**Comment by jackysp** [comment]
/run-all-tests tidb-test=release-2.1 tidb-private-test=release-2.1 tikv=release-2.1

**Comment by jackysp** [comment]
/run-all-tests tidb-test=release-2.1 tidb-private-test=release-2.1 tikv=release-2.1

**Comment by jackysp** [comment]
/run-all-tests tidb-test=release-2.1 tidb-private-test=release-2.1 tikv=release-2.1

**Comment by winkyao** [comment]
Please fix ci.

**Comment by zhouqiang-cl** [comment]
go: finding github.com/lysu/parser v0.0.0-20190325074808-d880cf39390b

go: github.com/lysu/parser@v0.0.0-20190325074808-d880cf39390b: unknown revision d880cf39390b

go: error loading module requirements
build failed

**Comment by jackysp** [comment]
/run-all-tests tidb-test=release-2.1 tidb-private-test=release-2.1 tikv=release-2.1

**Comment by zz-jason** [comment]
LGTM

File: executor/delete.go [hunk 1]
```diff
@@ -91,9 +92,9 @@ func (e *DeleteExec) deleteSingleTableByChunk(ctx context.Context) error {
 		break
 	}
 
-	// If tidb_batch_delete is ON and not in a transaction, we could use BatchDelete mode.
-	batchDelete := e.ctx.GetSessionVars().BatchDelete && !e.ctx.GetSessionVars().InTxn()
-	batchDMLSize := e.ctx.GetSessionVars().DMLBatchSize
+	sessVars := e.ctx.GetSessionVars()
+	// If tidb_batch_delete or enable-batch-dml is ON and not in a transaction, we could use BatchDelete mode.
```
Comment (zimulala) [inline executor/delete.go]: This comment doesn't quite match the code.
The comment like `tidb_batch_delete  && !sessVars.InTxn() ||  enable-batch-dml && !sessVars.InTxn()` or `(tidb_batch_delete  ||  enable-batch-dml) && !sessVars.InTxn()`.

Comment (jackysp) [reply executor/delete.go]: ah, yes.

File: executor/write_test.go [hunk 2]
```diff
@@ -2214,3 +2215,79 @@ func (s *testSuite) TestDeferConstraintCheckForInsert(c *C) {
...
+	tk1.MustExec("use test")
+	tk.MustExec("set @@session.tidb_dml_batch_size=2;")
+
+	tk.MustExec("begin")
+	tk1.MustExec("update t set j = 'e'")
+	_, err = tk.Exec("update t set j = 'd'")
+	c.Assert(executor.ErrBatchDMLFail.Equal(err), IsTrue, Commentf("error %s", err))
+	tk.MustQuery("select * from t order by i").Check(testkit.Rows("1 e", "2 e", "3 e"))
+
+	// Test normal retry
```
Comment (tiancaiamao) [inline executor/write_test.go]: "Test normal retry" makes me confused, is there a retry in the following code?

Comment (jackysp) [reply executor/write_test.go]: tk will retry at line 2291

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**